v-next/hardhat-keystore/src/internal/loaders/keystore-file-loader.ts

```diff
@@ -0,0 +1,88 @@
+import type {
+  FileManager,
+  Keystore,
+  KeystoreLoader,
+  UnencryptedKeystoreFile,
+} from "../types.js";
+
+import {
+  assertHardhatInvariant,
+  HardhatError,
+} from "@ignored/hardhat-vnext-errors";
+
+import { UnencryptedKeystore } from "../keystores/unencrypted-keystore.js";
+import { unencryptedKeystoreFileSchema } from "../types-validation.js";
+
+export class KeystoreFileLoader implements KeystoreLoader {
```

Comment:
The loader takes responsibility for:

- reading from the file - including validating the file
- writing back an updated keystore
- caching the keystore

Reply:
Maybe add this as a doc comment in its interface?